src/common/actions/repositories.js

```diff
@@ -53,8 +54,124 @@ export function fetchUserRepositories(pageNumber) {
 
 export function fetchRepositoriesError(error) {
   return {
-    type: FETCH_REPOSITORIES_ERROR,
+    type: REPOSITORIES_FAILURE,
     payload: error,
     error: true
   };
 }
+
+// XXX may need to split out the fetch from the success and failure dispatch
+// to ensure than setRepositories happens in sync with refreshed user snaps
+export function fetchUserRepositoriesAndSnaps(owner) {
+  return (dispatch) => {
+    return Promise.all([
+      dispatch(fetchUserRepositories()),
+      dispatch(fetchUserSnaps(owner))
+    ]);
+  };
+}
+
+// XXX move to repository actions?
+// could be plural or singular, could be in one module or two, might be clearer in one for now
+export const REPOSITORY_TOGGLE_SELECT = 'REPOSITORY_TOGGLE_SELECT';
+
+export function toggleRepositorySelection(id) {
+  return {
+    type: REPOSITORY_TOGGLE_SELECT,
+    payload: {
+      id
+    }
+  };
+}
+
+export const REPOSITORY_BUILD = 'REPOSITORY_BUILD';
+export const REPOSITORY_SUCCESS = 'REPOSITORY_SUCCESS';
+export const REPOSITORY_FAILURE = 'REPOSITORY_FAILURE';
+export const REPOSITORY_RESET = 'REPOSITORY_RESET';
+
+export function buildRepositories(repositories) {
+  return (dispatch) => {
+    const promises = repositories.map(
+      (repository) => {
+        return dispatch(buildRepository(repository));
+      }
+    );
+    return Promise.all(promises).then(() => dispatch(push('/dashboard')));
+  };
+}
+
+export function buildRepository(repository) {
+  const { id, url, owner, name } = repository;
+
+  return async (dispatch) => {
+    dispatch({
+      type: REPOSITORY_BUILD,
+      payload: {
+        id
+      }
+    });
+
+    try {
+      await createWebhook(owner, name);
+      const response = await fetch(`${BASE_URL}/api/launchpad/snaps`, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify({ repository_url: url }),
+        credentials: 'same-origin'
+      });
+
+      await checkStatus(response);
+      dispatch(buildRepositorySuccess(id));
+    } catch (error) {
+      dispatch(buildRepositoryError(id, error));
+      return Promise.reject(error);
```

Comment:
In this case it was probably meant to be resolved - error was handled (action dispatched) so no need to reject it.

I get your point that we may want to combine them, you may end up thinking they were successfuly resolved when they actually resulted in error.
But does the parent 'action' (that does Promise.all) need to know what was the result of child promises - I mean does it care if they were successes or failures or just that they finished?

Because each action results on it's own in state change, so error result is in state. So maybe we don't need to know if they failed or succeeded on 'parent' level, just that they finished fetching - and we can get the result from the state?